spec/unit/source/git_spec.rb

```diff
@@ -62,6 +62,38 @@
       expect(master_env.dirname).to eq 'master'
     end
   end
+
+  describe "generating environments without the ignore_branch_prefixes" do
+    before do
+      allow(subject.cache).to receive(:branches).and_return ['master', 'development', 'production', 'dev-test', 'dev', 'test_2']
+      subject.instance_variable_set(:@ignore_branch_prefixes, ['dev', 'test'])
+    end
+
+    let(:environments) { subject.generate_environments }
+
+    it "creates an environment for each branch not in ignore_branch_prefixes" do
+      expect(subject.generate_environments.size).to eq(2)
+    end
+
+    it "copies the source remote to the environment" do
+      expect(environments[0].remote).to eq subject.remote
+      expect(environments[1].remote).to eq subject.remote
+    end
+
+    it "uses the branch name as the directory by default" do
+      expect(environments[0].dirname).to eq 'master'
+      expect(environments[1].dirname).to eq 'production'
+    end
+  end
+
+  describe "filtering branches with ignore prefixes" do
+    let(:branches) { ['master', 'development', 'production', 'dev-test', 'dev', 'test_2'] }
```

Comment:
I think a test branch with dev or test at the end or the middle of a valid branch name would be good.

Reply:
Good idea.

Reply:
Adding this caused me to notice that "dev-test" is not a valid environment name, anyway.